babel/messages/frontend.py

```diff
@@ -401,6 +415,11 @@ def finalize_options(self):
             if not self.version:
                 self.version = self.distribution.get_version()
 
+        if self.add_location == 'never':
+            self.no_location = True
+        elif self.add_location == 'file':
+            self.include_lineno = False
+
```

Comment:
It might be worth it to `del self.add_location` here, as the variable has already served its purpose and further modifications to it will not have any effect.